x-pack/plugins/enterprise_search/public/applications/shared/layout/nav.tsx

```diff
@@ -15,11 +15,12 @@ import {
   ENTERPRISE_SEARCH_OVERVIEW_PLUGIN,
   WORKPLACE_SEARCH_PLUGIN,
 } from '../../../../common/constants';
+import { ProductAccess } from '../../../../common/types';
 import { SEARCH_INDICES_PATH } from '../../enterprise_search_content/routes';
 
 import { generateNavLink } from './nav_link_helpers';
 
-export const useEnterpriseSearchNav = () => {
+export const useEnterpriseSearchNav = (access: ProductAccess) => {
```

Comment:
@jgr There is a [test](https://github.com/elastic/kibana/blob/4d30305d94b86682caab249915ebf30f9c4e8d74/x-pack/plugins/enterprise_search/public/applications/shared/layout/nav.test.tsx#L15) of this function that will need to be updated with this PR. We should probably add a test for at least `NO_ACCESS` as well. IMO

Reply:
Yeah, just pushing up some new changes to double check I got everything, then I'll update the tests.